File: publisher.cpp
#include <iostream>
#include <string>
#include "publisher.h"
using namespace std;
int Publisher::publisher_count = 0;
// Конструктор без параметров
Publisher::Publisher() {
	name = "";
	address = "";
	phone_num = "";
	email = "";
	++publisher_count;
}
// Конструктор с 1 параметром
Publisher::Publisher(string name) {
	try {
		if (name.empty()) {
			throw invalid_argument("Название не может быть пустым");
		}
		this->name = name;
		++publisher_count;
	}
	catch (const invalid_argument& e) {
		cout << "Некорректный аргумент: " << e.what() << endl;
	}
}
// Конструктор со всеми параметрами
Publisher::Publisher(string name, string address, string phone_num, string email) {
	Publisher::setPublisherData(name, address, phone_num, email);
	++publisher_count;
}
// Сеттер
void Publisher::setPublisherData(string name, string address, string phone_num, string email) {
	this->name = name;
	this->address = address;
	this->phone_num = phone_num;
	this->email = email;
}
// Метод для пользовательского ввода данных
void Publisher::enterPublisherData() {
	cout << endl;
	cout << "Введите название издателя:" << endl;
	getline(cin, name);
	cout << "Введите адрес издателя:" << endl;
	getline(cin, address);
	cout << "Введите номер телефона издателя:" << endl;
	getline(cin, phone_num);
	cout << "Введите электронную почту издателя:" << endl;
	getline(cin, email);
	cout << endl;
	cout << "Издатель " << name << " был успешно создан!" << endl;
}
// Геттер
void Publisher::getPublisherData() {
	cout << endl;
	cout << "Информация об издателе:" << endl;
	cout << "Название: " << name << endl;
	cout << "Адрес: " << address << endl;
	cout << "Номер телефона: " << phone_num << endl;
	cout << "Электронная почта: " << email << endl;
}
// Статический геттер для получения кол-ва объектов
int Publisher::getPublisherCount() {
	return publisher_count;
}

File: src/com/github/Yazip/LR3/Publisher.java
package com.github.Yazip.LR3;
import java.util.Scanner;
// Класс издатель
public class Publisher {
    private String name;
    private String address;
    private String phone_num;
    private String email;
    private static int publisher_count = 0;
    // Конструктор без параметров
    public Publisher() {
        name = "";
        address = "";
        phone_num = "";
        email = "";
        ++publisher_count;
    }
    // Конструктор с 1 параметром
    public Publisher(String name) {
        try {
            if (name.isEmpty()) {
                throw new IllegalArgumentException("Название не может быть пустым");
            }
            this.name = name;
            ++publisher_count;
        }
        catch (IllegalArgumentException e) {
            System.out.println("Некорректный аргумент: " + e.getMessage());
        }
    }
    // Конструктор со всеми параметрами
    public Publisher(String name, String address, String phone_num, String email) {
        setPublisherData(name, address, phone_num, email);
        ++publisher_count;
    }
    // Сеттер
    public void setPublisherData(String name, String address, String phone_num, String email) {
        this.name = name;
        this.address = address;
        this.phone_num = phone_num.replaceAll("[^\\d]", "");
        this.email = email;
    }
    // Метод для пользовательского ввода данных
    public void enterPublisherData() {
        Scanner scanner = new Scanner(System.in, "UTF-8");
        System.out.println();
        System.out.println("Введите название издателя:");
        name = scanner.nextLine();
        System.out.println("Введите адрес издателя:");
        address = scanner.nextLine();
        System.out.println("Введите номер телефона издателя:");
        phone_num = scanner.nextLine();
        System.out.println("Введите электронную почту издателя:");
        email = scanner.nextLine();
        System.out.println();
        System.out.println("Издатель " + name + " был успешно создан!");
    }
    // Геттер
    public void getPublisherData() {
        System.out.println();
        System.out.println("Информация об издателе:");
        System.out.println("Название: " + name);
        System.out.println("Адрес: " + address);
        System.out.println("Номер телефона: " + phone_num);
        System.out.println("Электронная почта: " + email);
    }
    public static int getPublisherCount() {
        return publisher_count;
    }
}
